b2share/modules/upgrade/cli.py

```diff
@@ -0,0 +1,123 @@
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# B2Share is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with B2Share; if not, write to the Free Software Foundation, Inc.,
+# 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""B2Share cli commands for upgrades."""
+
+
+from __future__ import absolute_import, print_function
+
+from datetime import datetime
+
+import click
+from b2share.version import __version__
+from flask import current_app
+from flask.cli import with_appcontext, ScriptInfo
+from invenio_db import db
+from invenio_indexer.cli import index as index_cmd
+from subprocess import call
+
+from click.testing import CliRunner
+
+from .models import Migration
+
+
+@click.group()
+def upgrade():
+    """B2SHARE upgrade commands."""
+
+
+@upgrade.command()
+@with_appcontext
+@click.option('-v', '--verbose', is_flag=True, default=False)
+def run(verbose):
+    """Upgrade the database, activate alembic and reindex the records."""
+    alembic = current_app.extensions['invenio-db'].alembic
+    new_migration = Migration()
+    new_migration.version = __version__
+    new_migration.date = datetime.utcnow()
+    migration_data = {'elasticsearch': False,
+                      'alembic_version': False,
+                      'alembic_upgrade': False}
+
+    if (db.engine.dialect.has_table(db.engine, 'transaction') and
+            not alembic.migration_context._has_version_table()):
+
+        # Make sure alembic version table exists in the db
+        alembic.migration_context._ensure_version_table()
+
+        # alembic_version state of B2Share v2.0.1
+        heads_v2_0_1 = ['dbdbc1b19cf2', '1ba76da94103',
+                        '12a88921ada2', '97bbc733896c',
+                        '2f63be7b7572', 'e655021de0de',
+                        '999c62899c20', 'fb99eeaec4ac',
+                        '35d7d8958395']  # '35c1075e6360'
+
+        # Populate it as if alembic upgrade heads had ran
+        for head in heads_v2_0_1:
+            alembic.migration_context.stamp(
+                alembic.script_directory, head)
+        migration_data['alembic_version'] = True
+
+    # Upgrade if there are still outdated models
+    if verbose:
+        click.secho('Upgrading...', fg='green')
+    try:
+        alembic.upgrade()
+        click.secho('Done', fg='green')
+        migration_data['alembic_upgrade'] = True
+    except Exception as e:
+        pass
```

Comment:
TODO:
- rollback
- log the exception
- if possible commit the new_migration with the error message.
- stop the upgrade